core/middleware/md.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+
+var cheerio = require('cheerio');
+var prettyHrtime = require('pretty-hrtime');
+
+var marked = require('marked');
+var renderer = new marked.Renderer();
+marked.setOptions({
+  renderer: renderer
+});
+
+/*
+ * Get file content from response and parse contained Markdown markup
+ *
+ * @param {object} req - Request object
+ * @param {object} res - Response object
+ * @param {function} next - The callback function
+ * */
+exports.process = function (req, res, next) {
+    if (req.specData && req.specData.renderedHtml && req.specData.isMd) {
+        var start = process.hrtime();
+        var input = req.specData.renderedHtml;
+
+        // Processing with native markdown renderer
+        renderer.code = function (code, language) {
+            if (language === 'example') {
+                return '<div class="source_example">'+ code +'</div>';
+            } else if (language) {
+                return '<code class="src-'+ language +' source_visible">'+ code +'</code>';
+            } else {
+                return '<pre><code class="lang-source_wide-code">'+ code +'</code></pre>';
+            }
+        };
+
+        var $ = cheerio.load(marked(input));
+
+        // Spec description
+        var $H1 = $('h1');
```

Comment:
It does't seem to be the real overhead, strictly speaking. Anyway, it's up to you. I'm not quite sure in it.